src/DataCollector/ElasticaDataCollector.php

```diff
@@ -15,77 +15,154 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\Kernel;
```

Comment:
maybe it makes more sense to have 2 separate data collector classes one for legacy and one for the future? this way the code is cleaner to read and we can deprecate the legacy class

Reply:
It would make code cleaner, but it also means we have to change the class name, and ask users to explicitly switch to the new class when migrating to Symfony 5. This approach is not pretty but it does allow for a transparent upgrade from the user. Both solutions have pros and cons so, I don't know.